diesel_derives/src/util.rs

```diff
@@ -71,7 +71,7 @@ pub fn ty_for_foreign_derive(item: &DeriveInput, flags: &MetaItem) -> Result<Typ
 }
 
 pub fn fix_span(maybe_bad_span: Span, fallback: Span) -> Span {
-    let bad_span_debug = "Span(Span { lo: BytePos(0), hi: BytePos(0), ctxt: #0 })";
+    let bad_span_debug = "#0 bytes(0..0)";
```

Comment:
I think doing this is quite risky, because this may break at a minor version bump of `proc_macro2` 
Maybe there is some way to just construct a bad span itself? (This would at least make this code more robust against possible future changes of the debug output)

Reply:
This has less to do with `proc_macro2` and more to do with Rust. I agree that it's brittle and will break again in the future. That said, the only reason we do this at all is to work around a bug in Rust itself, and if this code breaks it just means that a handful of error cases point at the macro invocation instead.

I couldn't find any way to directly construct a bad span, no.